src/crs/projected/polarstereographic.jl

```diff
@@ -0,0 +1,203 @@
+# ------------------------------------------------------------------
+
+"""
+    PolarStereographicB{latF, lngₒ, Datum,Shift}
+
+Polar Stereographic CRS Variant B with latitude of standard parallel `latF` and  longitude of 
+origin `lngₒ` in degrees, `Datum` and `Shift`. Latitude of origin is taken to be ±90°,
+with the sign matching the sign of `latF`.
+
+See conversion formulas at [epsg.org](https://epsg.org/coord-operation-method_9829/Polar-Stereographic-variant-B.html)
+and in [EPSG guidance note #7-2 (pdf)](https://www.iogp.org/wp-content/uploads/2019/09/373-07-02.pdf).
+"""
+struct PolarStereographicB{latF,lngₒ,Datum,Shift,M<:Met} <: Projected{Datum,Shift}
+  x::M
+  y::M
+end
+
+PolarStereographicB{latF,lngₒ,Datum,Shift}(x::M, y::M) where {latF,lngₒ,Datum,Shift,M<:Met} =
+  PolarStereographicB{latF,lngₒ,Datum,Shift,float(M)}(x, y)
+PolarStereographicB{latF,lngₒ,Datum,Shift}(x::Met, y::Met) where {latF,lngₒ,Datum,Shift} =
+  PolarStereographicB{latF,lngₒ,Datum,Shift}(promote(x, y)...)
+PolarStereographicB{latF,lngₒ,Datum,Shift}(x::Len, y::Len) where {latF,lngₒ,Datum,Shift} =
+  PolarStereographicB{latF,lngₒ,Datum,Shift}(uconvert(m, x), uconvert(m, y))
+PolarStereographicB{latF,lngₒ,Datum,Shift}(x::Number, y::Number) where {latF,lngₒ,Datum,Shift} =
+  PolarStereographicB{latF,lngₒ,Datum,Shift}(addunit(x, m), addunit(y, m))
+
+PolarStereographicB{latF,lngₒ,Datum}(args...) where {latF,lngₒ,Datum} =
+  PolarStereographicB{latF,lngₒ,Datum,Shift()}(args...)
+
+PolarStereographicB{latF,lngₒ}(args...) where {latF,lngₒ} = PolarStereographicB{latF,lngₒ,WGS84Latest}(args...)
+
+Base.convert(
+  ::Type{PolarStereographicB{latF,lngₒ,Datum,Shift,M}},
+  coords::PolarStereographicB{latF,lngₒ,Datum,Shift}
+) where {latF,lngₒ,Datum,Shift,M} = PolarStereographicB{latF,lngₒ,Datum,Shift,M}(coords.x, coords.y)
+
+constructor(::Type{<:PolarStereographicB{latF,lngₒ,Datum,Shift}}) where {latF,lngₒ,Datum,Shift} =
+  PolarStereographicB{latF,lngₒ,Datum,Shift}
+
+lentype(::Type{<:PolarStereographicB{latF,lngₒ,Datum,Shift,M}}) where {latF,lngₒ,Datum,Shift,M} = M
+
+==(
+  coords₁::PolarStereographicB{latF,lngₒ,Datum,Shift},
+  coords₂::PolarStereographicB{latF,lngₒ,Datum,Shift}
+) where {latF,lngₒ,Datum,Shift} = coords₁.x == coords₂.x && coords₁.y == coords₂.y
+
+# ------------
+# CONVERSIONS
+# ------------
+
+#=
+Tested against the manual with:
+julia> convert(CoordRefSystems.PolarStereographicB{-71, 70, WGS84Latest, CoordRefSystems.Shift(xₒ=6e6m, yₒ=6e6m)}, LatLon(-75,120))
+┌ Debug: Values
+│   tF = 0.16840732452916302
+│   mF = 0.32654678137878085
+│   kO = 0.9727690128917972
+│   t = 0.13250834771195819
+│   ρ = 0.25693760394410403
+└ @ CoordRefSystems ~/.julia/dev/CoordRefSystems/src/crs/projected/polarstereographic.jl:120
+PolarStereographicB{WGS84Latest} coordinates with lonₒ: 0.0°, xₒ: 6.0e6 m, yₒ: 6.0e6 m
+├─ x: 7.255380793258386e6 m
+└─ y: 7.053389560610154e6 m
+=#
+# TODO: can FE and FN (false east and north) be removed by composing with a shift?
```

Comment:
Yes, the false easting and northing are treated externally along with lon0. The EPSG:3031 code can be mapped to `shift(PolarStereographic{...}, lon0=..., x=..., y=...)` and this function will create the shifted CRS.